### PR TITLE
[NeuralChat] Add askdoc retrieval api & example

### PR DESCRIPTION
**Pull request by letonghan**
## Type of Change

feature
API changed

## Description

Add askdoc retrieval api & example using neural chat API
JIRA: https://jira.devtools.intel.com/browse/NLPTOOLKIU-894

## Expected Behavior & Potential Risk

Askdoc backend will be able to start using this example
Included functions:
1. Chat
2. RAG
3. Upload files
4. Thumb up/down for a pair of Q&A
5. Export and download the like/dislike excel

## How has this PR been tested?

All functions are tested on local SPR server.

## Dependency Change?

No

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/examples/retrieval/Annual_Report.pdf
Comment (lvliang-intel) [inline intel_extension_for_transformers/neural_chat/examples/retrieval/Annual_Report.pdf:1]: Why delete this file?

File: intel_extension_for_transformers/neural_chat/examples/askgm/askgm.py
Comment (lvliang-intel) [inline intel_extension_for_transformers/neural_chat/examples/askgm/askgm.py:1]: don't name it 'askgm', rename it to 'askdocument'?

File: intel_extension_for_transformers/neural_chat/examples/retrieval/Annual_Report.pdf
Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/examples/retrieval/Annual_Report.pdf:1]: Deleted by mistake. Will recover it.

**Comment by hshen14** [comment]
Why need to mention askgm? I removed in the title.

**Review by hshen14 — APPROVED** [review]

**Pull request merged**